pulseaudio-control.bash

```diff
@@ -59,13 +61,44 @@ function getSinkName() {
 # "Sink #<index>" is used.
 function getNickname() {
     getSinkName "$1"
+    unset nickname
+
     if [ -n "${SINK_NICKNAMES[$sinkName]}" ]; then
         nickname="${SINK_NICKNAMES[$sinkName]}"
-    else
+    elif [ -n "$SINK_NICKNAME_PROP" ]; then
+        nickname="$(getNicknameFromProp "$SINK_NICKNAME_PROP" "$sinkName")"
+        # Cache that result for next time
+        SINK_NICKNAMES["$sinkName"]="$nickname"
+    fi
+
+    if [ -z "$nickname" ]; then
         nickname="Sink #$1"
     fi
 }
 
+# Gets sink nickname based on a given property.
+function getNicknameFromProp() {
+    local nickname_prop="$1"
+    local for_name="$2"
+
+    mapfile -t lines < <(pacmd list-sinks)
+    for line in "${lines[@]}"; do
+        case "$line" in
+            *name:*)
+                sink_name="$(echo "$line" | sed -E 's/.*name: <(.*)>/\1/')"
+                unset sink_desc
+                ;;
+            *"$nickname_prop ="*)
+                if [ "$sink_name" != "$for_name" ]; then
+                    continue
+                fi
+                prop_value="$(echo "$line" | sed -E "s/.*$nickname_prop = \"(.*)\"/\1/")"
+                echo "$prop_value"
```

Comment:
is this echo for debugging? you might want to remove it

Reply:
No, that's the found nickname - consumed on L69.

Reply:
Ah ok. Wouldn't it be better to directly return it in the `nickname` variable?

Reply:
I prefer to keep variables `local` to their function (and pretend I'm using a 'real' programming language :smile:) but I see that's the style used elsewhere, I'll change it for consistency.